src/main/java/com/couchbase/lite/util/Utils.java

```diff
@@ -282,6 +285,25 @@ public static boolean isGzip(String contentEncoding) {
         return map;
     }
 
+    public static String join(List<String> list, String separator) {
+        StringBuilder sb = new StringBuilder();
+        for (String str : list) {
+            if (sb.length() != 0)
+                sb.append(separator);
+            sb.append(str);
+        }
+        return sb.toString();
+    }
+
+    public static Map<String, String> headersToMap(Map<String, List<String>> headers) {
+        List<String>a = new ArrayList<String>();
+        Map<String, String> map = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
Just a question. Does it need to be in-order?

Reply:
https://tools.ietf.org/html/rfc2616#section-4.2
>The order in which header fields with differing field names are
   received is not significant. However, it is "good practice" to send
   general-header fields first, followed by request-header or response-
   header fields, and ending with the entity-header fields.

So not necessary to be alphabetical order.